lib/chef/provider/package/chocolatey.rb

```diff
@@ -238,6 +242,32 @@ def available_packages
           @available_packages
         end
 
+        # Getting packages from Artifactory server and return as candidate
+        def available_packages_from_artifactory
+          return @available_packages_from_artifactory if @available_packages_from_artifactory
+          @available_packages_from_artifactory = {}
+          package_name_array.length.times do |item|
+            available_versions =
+              begin
+                cmd = [ "list -r #{package_name_array[item]}" ]
+                cmd.push( "-source #{new_resource.source}" )
+                raw = parse_list_output(*cmd).first.each_slice(2).to_h
+                raw1 = parse_list_output(*cmd)
```

Comment:
can we fold this logic into `available_packages()`, so that we inspect the output of `list -r` to determine if we need to do the extra processing?  then there'd be no magic of inspecting the `source` property for "artifactory" and it would collapse it all down to one codepath.